src/TwigComponent/src/Twig/ComponentTokenParser.php

```diff
@@ -118,6 +118,14 @@ private function generateEmbeddedTemplateIndex(string $file, int $line): int
             $this->lineAndFileCounts[$fileAndLine] = 0;
         }
 
-        return crc32($fileAndLine).++$this->lineAndFileCounts[$fileAndLine];
+        $index = crc32($fileAndLine).++$this->lineAndFileCounts[$fileAndLine];
+
+        if (4 === \PHP_INT_SIZE) {
+            // On 32-bit PHP, the index can be negative
+            // we need to convert it to a positive integer
+            $index = abs($index) % \PHP_INT_MAX + 1;
```

Comment:
```suggestion
            // On 32-bit PHP, the index can be negative or greater than PHP_INT_MAX
            // we need to convert it to a positive 32-bit integer
            $index = fmod(abs($index), \PHP_INT_MAX) + 1;
```
Using `fmod` here instead of the modulo operator is important, because `abs($index)` might return a float greater than `PHP_INT_MAX`, which `%` would [convert to a negative integer](https://www.php.net/manual/en/language.operators.arithmetic.php#:~:text=Operands%20of%20modulo%20are%20converted%20to%20int), resulting [in a negative](https://www.php.net/manual/en/language.operators.arithmetic.php#:~:text=has%20the%20same%20sign%20as%20the%20dividend) remainder.

Reply:
FWIW, we could add explicit integer conversion to that line to spare confusion. Up to you.